public/app/plugins/datasource/postgres/datasource.test.ts

```diff
@@ -12,28 +12,44 @@ import {
   LoadingState,
   createDataFrame,
 } from '@grafana/data';
-import { FetchResponse } from '@grafana/runtime';
-import { backendSrv } from 'app/core/services/backend_srv'; // will use the version in __mocks__
+import {
+  BackendSrv,
+  DataSourceSrv,
+  FetchResponse,
+  getBackendSrv,
+  setBackendSrv,
+  getDataSourceSrv,
+  setDataSourceSrv,
+} from '@grafana/runtime';
 import { QueryFormat, SQLQuery } from 'app/features/plugins/sql/types';
 import { makeVariable } from 'app/features/plugins/sql/utils/testHelpers';
 
 import { PostgresDatasource } from './datasource';
 import { PostgresOptions } from './types';
 
-jest.mock('@grafana/runtime', () => ({
-  ...jest.requireActual('@grafana/runtime'),
-  getBackendSrv: () => backendSrv,
-}));
-
-jest.mock('@grafana/runtime/src/services', () => ({
-  ...jest.requireActual('@grafana/runtime/src/services'),
-  getBackendSrv: () => backendSrv,
-  getDataSourceSrv: () => {
-    return {
-      getInstanceSettings: () => ({ id: 8674 }),
-    };
-  },
-}));
+const backendSrv: BackendSrv = {
```

Comment:
Can you do the simpler version in here as well?

Reply:
unfortunately no 😿,it is not enough. if you check the "before" code, you can see that it had to mock `@grafana/runtime/src/services` too, and i don't even know if that would work from an external plugin.

the reason (i think, i'm like 80% sure) is that some test-call eventually reaches https://github.com/grafana/grafana/blob/main/packages/grafana-runtime/src/utils/DataSourceWithBackend.ts , and that thing imports `getBackendSrv` and `getDataSourceSrv` from `../services`, not from `@grafana/runtime` (because it itself is in `@grafana/runtime`), so the mock does not catch it.

Reply:
right, it is strange that it works in the mysql tests. Anyway let's merge this now and get rid of the marble testing later.